ui/component/commentReply/view.jsx

```diff
@@ -0,0 +1,88 @@
+// @flow
+import { CHANNEL_NEW } from 'constants/claim';
+import React, { useEffect, useState } from 'react';
+import { FormField, Form } from 'component/common/form';
+import Button from 'component/button';
+import ChannelSection from 'component/selectChannel';
+import usePersistedState from 'effects/use-persisted-state';
+import * as MODALS from 'constants/modal_types';
+import I18nMessage from '../i18nMessage/view';
+
+type Props = {
+  commentingEnabled: boolean,
+  claim: StreamClaim,
+  parentId: string,
+  openModal: (id: string, { onCommentAcknowledge: () => void }) => void,
+  replyComment: (string, string, string, string) => void,
+  setReplying: boolean => void,
+};
+
+export function CommentReply(props: Props) {
+  const { commentingEnabled, replyComment, setReplying, claim, openModal, parentId } = props;
+  const { claim_id: claimId } = claim;
+  const [commentValue, setCommentValue] = usePersistedState(`comment-${claimId}`, '');
+  const [commentAck, setCommentAck] = usePersistedState('comment-acknowledge', false);
+  const [channel, setChannel] = usePersistedState('comment-channel', 'anonymous');
+  const [charCount, setCharCount] = useState(commentValue.length);
+
+  function handleCommentChange(event) {
+    setCommentValue(event.target.value);
+  }
+
+  function handleChannelChange(channel) {
+    setChannel(channel);
+  }
+
+  function handleCommentAck() {
+    setCommentAck(true);
+  }
+
+  function onTextareaFocus() {
+    if (!commentAck) {
+      openModal(MODALS.COMMENT_ACKNOWEDGEMENT, { onCommentAcknowledge: handleCommentAck });
+    }
+  }
+
+  function handleSubmit() {
+    if (channel !== CHANNEL_NEW && commentValue.length);
+    replyComment(commentValue, claimId, channel, parentId);
+    setCommentValue('');
+  }
+
+  useEffect(() => setCharCount(commentValue.length), [commentValue]);
+
+  if (!commentingEnabled) {
+    return (
+      <I18nMessage tokens={{ sign_in_link: <Button button="link" requiresAuth label={__('sign in')} /> }}>
+        Please %sign_in_link% to comment.
+      </I18nMessage>
+    );
+  }
+
+  return (
+    <Form className="comment__reply-form" onSubmit={handleSubmit}>
```

Comment:
I still think this component should be combined with `CommentCreate`. They are mostly the same.

Reply:
Any proposed way that I should do that? If I were to combine it into the CommentCreate component I feel like it would get very cluttered in CommentCreate when we could just keep it a separate component. There are some styles and functionality specific to replying, it seems counterproductive to put them all in one class. I feel it is neater this [way](https://en.wikipedia.org/wiki/Single_responsibility_principle).

Reply:
The only big difference between the two is calling `replyComment` instead of `createComment`. You could pass in a single prop called `isReply` that adds the cancel button, and changes what function is called when a user submits the form.